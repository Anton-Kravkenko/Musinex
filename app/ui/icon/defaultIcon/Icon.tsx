import { usePressAnimation } from '@/animation/usePressAnimation'
import { IconType, UPressableProps } from '@/types/global'
import { getHexCode } from '@/utils/getColor'
import { Ionicons } from '@expo/vector-icons'
import { useColorScheme } from 'nativewind'
import { FC, memo } from 'react'
import { Pressable } from 'react-native'
import Animated from 'react-native-reanimated'
import { theme } from '../../../../tailwind.config'

interface IconProps extends UPressableProps {
	name: IconType
	color?: keyof typeof theme.colors
	size?: number
	border?: boolean
	backgroundColor?: string
	borderRadius?: number
	padding?: number
}

const UIcon: FC<IconProps> = ({
	name,
	color,
	size = 24,
	border = false,
	backgroundColor = 'transparent',
	borderRadius = 16,
	padding = 8,
	...props
}) => {
	const { colorScheme } = useColorScheme()
	const { animatedStyle, pressFunctions } = usePressAnimation()
	return (
		<Pressable pointerEvents={'auto'} {...pressFunctions} {...props}>
			<Animated.View
				style={[
					{
						justifyContent: 'center',
						alignItems: 'center',
						padding: padding,
						borderWidth: border ? 2 : 0,
						borderColor:
							colorScheme === 'light'
								? getHexCode('dark')
								: getHexCode('lightBlack'),
						backgroundColor: backgroundColor,
						borderRadius: borderRadius
					},
					animatedStyle
				]}
			>
				<Ionicons
					name={name}
					color={
						color
							? getHexCode(color)
							: colorScheme === 'light'
							? getHexCode('dark')
							: getHexCode('white')
					}
					size={size}
				/>
			</Animated.View>
		</Pressable>
	)
}

export default memo(UIcon)
