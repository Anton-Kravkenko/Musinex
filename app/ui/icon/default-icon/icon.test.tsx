import Icon from '@/ui/icon/default-icon/icon'
import { cleanup, fireEvent, render } from '@testing-library/react-native'

const mockFn = jest.fn()
afterEach(() => cleanup())
describe('Icon', () => {
	test('should render currect', () => {
		const renders = render(
			<Icon
				name={'search'}
				style={{
					width: 20
				}}
				className='mb-5'
			/>
		)
		expect(renders).toMatchSnapshot()
	})
	test('press', () => {
		const { getByTestId } = render(
			<Icon
				name={'search'}
				style={{
					width: 20
				}}
				className='mb-5'
				onPress={mockFn}
			/>
		)
		fireEvent.press(getByTestId('icon-wrapper'))
		expect(mockFn).toHaveBeenCalledTimes(1)
	})
})
