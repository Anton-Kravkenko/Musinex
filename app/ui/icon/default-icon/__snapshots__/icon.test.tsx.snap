// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render currect 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  className="mb-5"
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="auto"
  style={
    {
      "width": 20,
    }
  }
  testID="icon-wrapper"
>
  <Text />
</View>
`;
